tokio/src/sync/semaphore.rs

```diff
@@ -676,6 +676,11 @@ impl OwnedSemaphorePermit {
         self.permits += other.permits;
         other.permits = 0;
     }
+
+    /// Returns the [`Semaphore`] from which this permit was acquired.
+    pub fn semaphore(&self) -> Arc<Semaphore> {
+        self.sem.clone()
+    }
```

Comment:
The user can clone it themselves if they need a clone.
```suggestion
    /// Returns the [`Semaphore`] from which this permit was acquired.
    pub fn semaphore(&self) -> &Arc<Semaphore> {
        &self.sem
    }
```

Reply:
Good idea!